x/wasm/keeper/keeper.go

```diff
@@ -1081,10 +1155,19 @@ func (k Keeper) importContract(ctx sdk.Context, contractAddr sdk.AccAddress, c *
 		return err
 	}
 
-	k.appendToContractHistory(ctx, contractAddr, entries...)
-	k.storeContractInfo(ctx, contractAddr, c)
-	k.addToContractCodeSecondaryIndex(ctx, contractAddr, entries[len(entries)-1])
-	k.addToContractCreatorSecondaryIndex(ctx, creatorAddress, entries[0].Updated, contractAddr)
+	err = k.appendToContractHistory(ctx, contractAddr, entries...)
+	if err != nil {
+		return err
+	}
+	k.mustStoreContractInfo(ctx, contractAddr, c)
+	err = k.addToContractCodeSecondaryIndex(ctx, contractAddr, entries[len(entries)-1])
```

Comment:
This comment is not related to this particular pr, but it has happened during tgrade integrations that entries were not initialized and it would panic at this point because of IndexOutOfRange. Do you think it would make sense to add a check and panic with a meaningful error?